pkg/asset/installconfig/powervs/metadata.go

```diff
@@ -282,3 +285,77 @@ func (m *Metadata) GetDNSServerIP(ctx context.Context, vpcName string) (string,
 	}
 	return dnsServerIP, nil
 }
+
+// CreateDNSRecord creates a CNAME record for the specified hostname and destination hostname.
+func (m *Metadata) CreateDNSRecord(ctx context.Context, hostname string, destHostname string) error {
+	instanceCRN, err := m.client.GetInstanceCRNByName(ctx, m.BaseDomain, m.PublishStrategy)
+	if err != nil {
+		return fmt.Errorf("failed to get InstanceCRN (%s) by name: %w", m.PublishStrategy, err)
+	}
+
+	backoff := wait.Backoff{
+		Duration: 15 * time.Second,
+		Factor:   1.1,
+		Cap:      leftInContext(ctx),
+		Steps:    math.MaxInt32,
+	}
+
+	err = wait.ExponentialBackoffWithContext(ctx, backoff, func(context.Context) (bool, error) {
+		err2 := m.client.CreateDNSRecord(ctx, m.PublishStrategy, instanceCRN, m.BaseDomain, hostname, destHostname)
+		if err2 == nil {
+			return true, nil
+		}
+		return false, err2
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create a DNS CNAME record (%s, %s): %w",
+			hostname,
+			destHostname,
+			err)
+	}
+	return err
+}
+
+// ListSecurityGroupRules lists the rules created in the specified VPC.
+func (m *Metadata) ListSecurityGroupRules(ctx context.Context, vpcID string) (*vpcv1.SecurityGroupRuleCollection, error) {
+	return m.client.ListSecurityGroupRules(ctx, vpcID)
+}
+
+// SetVPCServiceURLForRegion sets the URL for the VPC based on the specified region.
+func (m *Metadata) SetVPCServiceURLForRegion(ctx context.Context, vpcRegion string) error {
+	return m.client.SetVPCServiceURLForRegion(ctx, vpcRegion)
+}
+
+// AddSecurityGroupRule adds a security group rule to the specified VPC.
+func (m *Metadata) AddSecurityGroupRule(ctx context.Context, rule *vpcv1.SecurityGroupRulePrototype, vpcID string) error {
+	backoff := wait.Backoff{
+		Duration: 15 * time.Second,
+		Factor:   1.1,
+		Cap:      leftInContext(ctx),
+		Steps:    math.MaxInt32,
+	}
+
+	err := wait.ExponentialBackoffWithContext(ctx, backoff, func(context.Context) (bool, error) {
+		err2 := m.client.AddSecurityGroupRule(ctx, vpcID, rule)
+		if err2 == nil {
+			return true, nil
+		}
+		return false, err2
+	})
```

Comment:
If you're not adding any retry logic:
```suggestion
	err := m.client.AddSecurityGroupRule(ctx, vpcID, rule)
```